k9mail/src/main/java/com/fsck/k9/message/PgpMessageBuilder.java

```diff
@@ -221,7 +301,9 @@ public void writeTo(OutputStream os) throws IOException {
                 if (currentState == State.OPENPGP_SIGN) {
                     mimeBuildSignedMessage(bodyPart, result);
                 } else if (currentState == State.OPENPGP_ENCRYPT) {
-                    mimeBuildEncryptedMessage(encryptedTempBody, result);
+                    mimeBuildEncryptedMessage(pgpResultTempBody, result);
+                } else if (currentState == State.OPENPGP_INLINE) {
+                    mimeBuildInlineMessage(pgpResultTempBody, result);
                 } else {
```

Comment:
inline can be either or both. in both cases though, it is a single pgp operation replacing the plain text of the mail with its pgp clearsigned/encrypted version, so it doesn't need two passes like sign/encrypt.